server/src/com/thoughtworks/go/config/GoConfigDao.java

```diff
@@ -179,7 +179,7 @@ public String md5OfConfigFile() {
     }
 
     public void updatePipeline(PipelineConfig pipelineConfig, LocalizedOperationResult result, Username currentUser, PipelineConfigService.SaveCommand saveCommand) {
-        synchronized (writeLock) {
+        synchronized (goConfigWriteLock.mutex()) {
             if (saveCommand.hasWritePermissions()) {
```

Comment:
As an aside, shouldn't this if clause have an else? I'd also add some logging, just like in updateConfig. Could help us see it wait to get this lock. Again, both are not related to this issue, I know.